### PR TITLE
turbo mode for lookup plugins

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

New turbo module to support lookup plugin

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: plugins/module_utils/turbo/lookup_server.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import argparse
```
Comment (goneri) [inline plugins/module_utils/turbo/lookup_server.py:1]: Do we need to create another server? I was under the impression it was possible to reuse the same for both module and lookup plugins.

File: plugins/module_utils/turbo/lookup.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import json
```
Comment (goneri) [inline plugins/module_utils/turbo/lookup.py:1]: This is a a but of duplicated code in this one. Can we move that in a shared module_utils file?

**Comment by goneri** [comment]
@abikouo this looks promising, nice :-).

**Comment by abikouo** [comment]
> @abikouo this looks promising, nice :-).

Thanks for the feedback. I will address your remark

File: plugins/module_utils/turbo/lookup.py (hunk 2 above)
Comment (abikouo) [reply plugins/module_utils/turbo/lookup.py:1]: I will do that

File: plugins/module_utils/turbo/lookup_server.py (hunk 1 above)
Comment (abikouo) [reply plugins/module_utils/turbo/lookup_server.py]: merged into a single server

**Comment by abikouo** [comment]
recheck

File: plugins/module_utils/turbo/lookup.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import os
```
Comment (goneri) [inline plugins/module_utils/turbo/lookup.py:1]: This module won't be used by any module, this is the reason why it should be in plugin_utils, not module_utils.

See: https://github.com/ansible/ansible/pull/75254#issuecomment-880099069

**Review by goneri — CHANGES_REQUESTED** [review]

File: plugins/lookup/turbo_demo.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Copyright: (c) 2021, Aubin Bikouo (@abikouo)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
```
Comment (goneri) [inline plugins/lookup/turbo_demo.py:4]: The two lines are Py2 specific. You can safely remove them.

**Review by goneri — CHANGES_REQUESTED** [review]

**Review by goneri — APPROVED** [review]

File: plugins/module_utils/turbo/server.py [hunk 5]
```diff
@@ -191,6 +191,41 @@ class FakeStdin:
             )
 
 
+def run_lookup_plugin(data):
+    errors = None
+    try:
+        import ansible.plugins.loader as plugin_loader
+        from ansible.parsing.dataloader import DataLoader
+        from ansible.template import Templar
```
[the remark this answers is not in the document]
Comment (abikouo) [reply plugins/module_utils/turbo/server.py:199]: Yes this has to be moved, but we agreed to do it in a separate PR

[the remark this answers is not in the document]
Comment (gravesm) [reply plugins/module_utils/turbo/server.py:199]: Ah, got it.

**Comment by abikouo** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Pull request merged**